src/arithmetic.c

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file arithmetic.c
+ * @author David Chung (david.chung1719@outlook.com)
+ * @brief Defines the different arithmetic operations used in main.c
+ * @version 3.0
+ * @date 02Dec24
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#include <stdint.h>
+#include "arithmetic.h"
+#include "debug.h"
+
+int32_t add(int32_t x, int32_t y)
+{
+    // Check for overflow
+    if (0 < x && 0 < y && (INT32_MAX - y) < x) 
```

Comment:
Make all comparisons explicitly bracketed 

```suggestion
    if ((0 < x) && (0 < y) && ((INT32_MAX - y) < x))
```